app/uk/gov/hmrc/cgtpropertydisposals/service/returns/ReturnsService.scala

```diff
@@ -113,6 +123,79 @@ class DefaultReturnsService @Inject() (
     } yield returnResponse
   }
 
+  private def deltaCharge(
+    response: DesSubmitReturnResponse,
+    cgtReference: CgtReference,
+    fromDate: LocalDate,
+    toDate: LocalDate
+  )(implicit hc: HeaderCarrier): EitherT[Future, Error, Option[DeltaCharge]] =
+    response.ppdReturnResponseDetails.chargeReference match {
+      case Some(chargeReference) =>
+        lazy val charge = for {
+          desFinancialData <- getDesFinalcialData(cgtReference, fromDate, toDate)
+          charge           <- EitherT.fromEither[Future](findDeltaCharge(desFinancialData.financialTransactions, chargeReference))
+        } yield charge
+
+        charge
+      case _                     => EitherT.pure(None)
+    }
+
+  private def getDesFinalcialData(
```

Comment:
this is the same code that appears elsewhere - it'd be better to make the original code not private and reuse it here

Reply:
yes right, I had same plan to write a function and reuse at both places but forgot that one.